test/integration/analytical_policy.hpp

```diff
@@ -159,8 +159,10 @@ void test_analytical_troe(BuilderPolicy& builder)
   {
     times.push_back(time_step);
     solver.CalculateRateConstants(state);
+    prepare_for_solve(state);
     // Model results
     auto result = solver.Solve(time_step, state);
+    postpare_for_solve(state);
```

Comment:
These are functions that wrap a single line of code. I really think moving the data onto and off of the device should be done by the solver when it needs to.

Reply:
I had a similar discussion with Matt last Friday about this and he made a good point: if we run MICM in a CPU-only host model like CAM, then the data movement here is necessary to ensure correctness; if we run MICM in a fully GPU-enabled model like FastEddy, we actually do not need any data movement here. Thus it should really be the user's liability/obligation to decide when the data movement needs to happen. Hiding the data movement inside the `solver` may lead to unnecessary data movement and hurt the performance later.

Reply:
Ah, I hadn't thought of that